src/cargo/core/profiles.rs

```diff
@@ -753,22 +750,22 @@ pub enum DebugInfo {
     /// faster to build (see [DebugInfo::weaken]).
     ///
     /// In all other situations, this level value will be the one to use.
-    Deferred(u32),
+    Deferred(TomlDebugInfo),
 }
 
 impl DebugInfo {
     /// The main way to interact with this debuginfo level, turning it into an Option.
-    pub fn to_option(&self) -> Option<u32> {
+    pub fn to_option(self) -> Option<TomlDebugInfo> {
         match self {
             DebugInfo::None => None,
-            DebugInfo::Explicit(v) | DebugInfo::Deferred(v) => Some(*v),
+            DebugInfo::Explicit(v) | DebugInfo::Deferred(v) => Some(v),
         }
     }
 
     /// Returns true if the debuginfo level is high enough (at least 1). Helper
     /// for a common operation on the usual `Option` representation.
     pub(crate) fn is_turned_on(&self) -> bool {
-        self.to_option().unwrap_or(0) != 0
+        !matches!(self.to_option(), None | Some(TomlDebugInfo::None))
```

Comment:
Does the comment for this function need to be updated?

Reply:
Good catch, done :)